modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -2561,7 +2561,8 @@ private void processSymbolizers(final Graphics2D graphics,
                     Geometry g = clipper.clipSafe(shape.getGeometry(), preserveTopology, 1);
                     
                     // handle perpendincular offset as needed
-                    if(style instanceof LineStyle2D && ((LineStyle2D) style).getPerpendicularOffset() != 0) {
+                    if(style instanceof LineStyle2D && ((LineStyle2D) style).getPerpendicularOffset() != 0
+                            && g != null && !g.isEmpty()) {
```

Comment:
No problem with the additional safety checks.

Is this a case of ``clipSafe(shape.getGeometry(), preserveTopology, 1)`` api not being respected? If so please update the javadocs on the method with a clarification.